src/spurt/workflows/emcf/_solver.py

```diff
@@ -248,22 +250,19 @@ def unwrap_gradients_in_space(self, grad_space: np.ndarray) -> np.ndarray:
         logger.info(f"Spatial: Number of links: {self.nlinks}")
         logger.info(f"Spatial: Number of cycles: {self._solver_space.ncycles}")
 
-        for ii in range(self.nifgs):
-            logger.info(f"Spatial: Unwrapping {ii + 1} / {self.nifgs}")
-            # Slice per ifg
-            ifg_grad = grad_space[ii, :]
-
-            # Compute residues
-            residues = self._solver_space.compute_residues_from_gradients(ifg_grad)
-
-            # Unwrap the interferogram - sequential
-            flows = self._solver_space.residues_to_flows(residues, cost)
-
-            # Flood fill
-            uw_data[ii, :] = utils.flood_fill(
-                ifg_grad, self._solver_space.edges, flows, mode="gradients"
-            )
-
+        with ProcessPoolExecutor(max_workers=self.settings.worker_count) as executor:
```

Comment:
Do we need an `mp_context=mp.get_context('fork')` as an argument to `ProcessPoolExecutor` here? Else, on OSX it might pick spawn which could be slow

Reply:
Hmm I hadn't noticed slowness when testing with spawn on my laptop in this case since the startup time is so much smaller than each work function's time (unlike the many temporal calls)

I had left unset here so that users could call [`set_start_method`](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.set_start_method) for themselves, e.g. so that a process with threads doesn't yell about you using `fork`.

Any opinion @gmgunter ?

Reply:
Unless there's a compelling reason to force the method to `'fork'` then I agree it seems like it'd be nice to keep it flexible and give the user control over the start method.

So are you saying that the reason the temporal unwrapping seems to be so sensitive to the start method is because each unit of parallel work is so small? So the startup time is a significant fraction of the runtime of each process?

That makes sense -- in that case we should be considering batching up the work so that each process has more stuff to do and the overhead is better amortized. [`imap_unordered()`](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool.imap_unordered) has an optional `chunksize` argument that we could play around with. It might be worth a nice speedup as well.